modules/globebrowsing/src/geojson/geojsoncomponent.cpp

```diff
@@ -0,0 +1,743 @@
+
+        // [[codegen::verbatim(CoordinateOffsetInfo.description)]]
+        std::optional<glm::vec2> coordinateOffset;
+
+        enum class [[codegen::map(openspace::globebrowsing::GlobeGeometryFeature::PointRenderMode)]] PointRenderMode {
+            AlignToCameraDir,
+            AlignToCameraPos,
+            AlignToGlobeNormal,
+            AlignToGlobeSurface
+        };
+        // [[codegen::verbatim(PointRenderModeInfo.description)]]
+        std::optional<PointRenderMode> pointRenderMode;
+
+        // [[codegen::verbatim(DrawWireframeInfo.description)]]
+        std::optional<bool> drawWireframe;
+
+        // These properties will be used as default values for the geoJson rendering,
+        // meaning that they will be used when there is no value given for the
+        // individual geoJson features
+        std::optional<ghoul::Dictionary> defaultProperties
+            [[codegen::reference("globebrowsing_geojsonproperties")]];
+
+        // [[codegen::verbatim(LightSourcesInfo.description)]]
+        std::optional<std::vector<ghoul::Dictionary>> lightSources
+            [[codegen::reference("core_light_source")]];
+    };
+#include "geojsoncomponent_codegen.cpp"
+
+
+    constexpr openspace::properties::Property::PropertyInfo FlyToFeatureInfo = {
+        "FlyToFeature",
+        "Fly To Feature",
+        "Triggering this leads to the camera flying to a position that show the GeoJSON "
+        "feature. The flight will account for any lat, long or height offset"
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo CentroidCoordinateInfo = {
+        "CentroidCoordinate",
+        "Centroid Coordinate",
+        "The lat long coordinate of the centroid position of the read geometry. Note "
+        "that this value does not incude the offset"
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo BoundingBoxInfo = {
+        "BoundingBox",
+        "Bounding Box",
+        "The lat long coordinates of the lower and upper corner of the bounding box of "
+        "the read geometry. Note that this value does not incude the offset"
+    };
+
+} // namespace
+
+namespace openspace::globebrowsing {
+
+documentation::Documentation GeoJsonComponent::Documentation() {
+    return codegen::doc<Parameters>("globebrowsing_geojsoncomponent");
+}
+
+GeoJsonComponent::SubFeatureProps::SubFeatureProps(
+                                       properties::PropertyOwner::PropertyOwnerInfo info)
+    : properties::PropertyOwner(info)
+    , enabled(EnabledInfo, true)
+    , flyToFeature(FlyToFeatureInfo)
+    , centroidLatLong(
+        CentroidCoordinateInfo,
+        glm::vec2(0.f),
+        glm::vec2(-90.f, -180.f),
+        glm::vec2(90.f, 180.f)
+    )
+    , boundingboxLatLong(
+        BoundingBoxInfo,
+        glm::vec4(0.f),
+        glm::vec4(-90.f, -180.f, -90.f, -180.f),
+        glm::vec4(90.f, 180.f, 90.f, 180.f)
+    )
+{
+    // From fadeable
+    addProperty(_opacity);
+    addProperty(_fade);
```

Comment:
Should we do this for all the other places as well then? 🤔

Reply:
Yea, we probably should, since you are totally right in your comment.  Feel free to resolve this when you read it;  i'll make those changes straight away